include/quicly/ranges.h

```diff
@@ -41,11 +47,29 @@ typedef struct st_quicly_ranges_t {
     quicly_range_t _initial;
 } quicly_ranges_t;
 
+/**
+ * initializes the structure
+ */
 static void quicly_ranges_init(quicly_ranges_t *ranges);
+/**
+ * initializes the structure, registering given range
+ */
 int quicly_ranges_init_with_range(quicly_ranges_t *ranges, uint64_t start, uint64_t end);
+/**
+ * clears the structure
+ */
 static void quicly_ranges_clear(quicly_ranges_t *ranges);
+/**
+ * adds given range, returns 0 if successful
+ */
 int quicly_ranges_add(quicly_ranges_t *ranges, uint64_t start, uint64_t end);
+/**
+ * subtracts given range, returns 0 if sucessful
+ */
 int quicly_ranges_subtract(quicly_ranges_t *ranges, uint64_t start, uint64_t end);
```

Comment:
"remove" works for me, though I would like to see if you think that would be consistent with the other function names.

FWIW, my view is that these functions define set operations. `quicly_ranges_add` function modifies the object so that it would become an union of the current object and the given set. `quicly_ranges_subtract` "subtracts" given set from the object. It is my understanding that we sometimes use "A-B" when referring to the latter operation (see https://en.wikipedia.org/wiki/Complement_(set_theory)#Definition_2), I also know that Oracle uses "MINUS" as a verb to refer to this operation.

All that said, I am happy to change the name to whatever else, assuming that it would not feel unnatural when we view this as a set operation.

Reply:
I understand -- subtract is fine. Can you add a comment that the parameters are start and end packet numbers?

Reply:
Actually they are not always. For example, they could be byte offsets of a stream, when `quicly_ranges_t` is used to represent the byte ranges that have been acknowledged.